vtex/loaders/promotion/getPromotionById.ts

```diff
@@ -0,0 +1,33 @@
+import { AppContext } from "../../mod.ts";
+import type { Document } from "../../utils/types.ts";
+import { parseCookie } from "../../utils/vtexId.ts";
+
+interface Props {
+  /**
+   * @description Promotion ID.
+   */
+  idCalculatorConfiguration: string;
+}
+
```

Comment:
```suggestion

export const defaultVisibility = 'private' 

```

This loader access a private route, so its not safe make this loader do be called in client, this key will make safe once you only can call on server using the ctx.invoke.

You could take a look in this [PR](https://github.com/deco-cx/deco/pull/884) to understand all possibilities